etcd-client/src/kv/mod.rs

```diff
@@ -175,18 +203,55 @@ impl Kv {
                             }
                         }
                         Err(e) => {
-                            warn!("Watch response contains error, the error is: {}", e);
-                            this_clone2.cache_err.store(true, Ordering::Release);
-                            break;
+                            warn!(
+                                "Watch response contains error, the error is: {}, clean cache",
+                                e
+                            );
+                            this_clone2.clean_cache().await;
+                            continue;
                         }
                     }
                 }
             })
             .detach();
         };
-
         this
     }
+    /// Clean cache
+    #[inline]
+    async fn clean_cache(&self) {
+        self.cache_availability.write().await;
+        self.cache.clean().await;
+    }
+}
+
+#[async_trait]
+impl Shutdown for Arc<KvInner> {
+    async fn shutdown(&mut self) -> Res<()> {
+        if self.cache_enable {
+            self.shutdown.send(()).await?;
+            self.shutdown.close();
+        }
+        Ok(())
+    }
+}
+
+impl Kv {
+    /// Creates a new `KvClient`.
+    ///
+    /// This method should only be called within etcd client.
+    pub(crate) fn new(
+        client: KvClient,
+        watch_client: WatchClient,
+        cache_size: usize,
+        cache_enable: bool,
+    ) -> Self {
+        Self {
+            inner: Arc::new(Lazy::new(move || {
```

Comment:
Why do we need Lazy here?

Reply:
If users don't use kv-client like watch, lock etc, don't create smol task. Lazy will create kv client when users use it first time.